src/miner.cpp

```diff
@@ -96,8 +96,8 @@ void BlockAssembler::resetBlock()
     nFees = 0;
 }
 
-Optional<int64_t> BlockAssembler::m_last_block_num_txs{nullopt};
-Optional<int64_t> BlockAssembler::m_last_block_weight{nullopt};
+std::optional<int64_t> BlockAssembler::m_last_block_num_txs{std::nullopt};
```

Comment:
Not required. `std::optional<>` will default initialize to `std::nullopt`. These lines can be removed entirely.

Reply:
They are static, but this should still zero initialize: https://en.cppreference.com/w/cpp/language/zero_initialization

Reply:
Does this seem right to you, @MarcoFalke:

```diff
diff --git a/src/miner.cpp b/src/miner.cpp
index 56543e743e..5239836e0f 100644
--- a/src/miner.cpp
+++ b/src/miner.cpp
@@ -96,8 +96,8 @@ void BlockAssembler::resetBlock()
     nFees = 0;
 }
 
-std::optional<int64_t> BlockAssembler::m_last_block_num_txs{std::nullopt};
-std::optional<int64_t> BlockAssembler::m_last_block_weight{std::nullopt};
+/* std::optional<int64_t> BlockAssembler::m_last_block_num_txs{std::nullopt}; */
+/* std::optional<int64_t> BlockAssembler::m_last_block_weight{std::nullopt}; */
 
 std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& scriptPubKeyIn)
 {
diff --git a/src/miner.h b/src/miner.h
index 286a8c693e..9dadbf1507 100644
--- a/src/miner.h
+++ b/src/miner.h
@@ -159,8 +159,8 @@ public:
     /** Construct a new block template with coinbase to scriptPubKeyIn */
     std::unique_ptr<CBlockTemplate> CreateNewBlock(const CScript& scriptPubKeyIn);
 
-    static std::optional<int64_t> m_last_block_num_txs;
-    static std::optional<int64_t> m_last_block_weight;
+    inline static std::optional<int64_t> m_last_block_num_txs{};
+    inline static std::optional<int64_t> m_last_block_weight{};
 
 private:
     // utility functions
```

Reply:
Is the `inline` needed? Any version that passes mining_basic.py under valgrind should be correct. See fa178a6385bf300499fb18940051fc4142fb5b6b

Reply:
I'm pretty sure inline is required:

```
In file included from miner.cpp:6:
./miner.h:162:35: error: non-const static data member must be initialized out of line
    static std::optional<int64_t> m_last_block_num_txs{};
                                  ^~~~~~~~~~~~~~~~~~~~~~
./miner.h:163:35: error: non-const static data member must be initialized out of line
    static std::optional<int64_t> m_last_block_weight{};
                                  ^~~~~~~~~~~~~~~~~~~~~
2 errors generated.
```

Inline member variables have been possible since c++17: https://www.codingame.com/playgrounds/2205/7-features-of-c17-that-will-simplify-your-code/inline-variables.

Reply:
thanks. Seems fine to change this to `inline static` initialization.